From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: CodeTheDev <code@codeslab.ca>
Date: Wed, 13 Jul 2022 15:11:04 -0400
Subject: [PATCH] Fix Paper 8019 out of order chat kick

Disabled kick from server if "sent out-of-order chat" is triggered.
This will most likely be a temporary patch until a solution is implemented
on either Paper or Velocity.

diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 1581215f7e8348e40d248c8e3d5ef43143e69579..8169e04c0986895ed07716cf1f965ab6ca232f9e 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -2167,17 +2167,20 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
     private boolean tryHandleChat(String s, Instant instant) {
         if (!this.updateChatOrder(instant)) {
             ServerGamePacketListenerImpl.LOGGER.warn("{} sent out-of-order chat: '{}'; {} > {}", this.player.getName().getString(), s, this.lastChatTimeStamp.get().getEpochSecond(), instant.getEpochSecond()); // Paper
-            this.server.scheduleOnMain(() -> { // Paper - push to main
-            this.disconnect(Component.translatable("multiplayer.disconnect.out_of_order_chat"));
-            }); // Paper - push to main
-            return false;
+        // Obamium start - temporary fix for PaperMC/Paper#8019 out of order chat kick
+            /*
+             * this.server.scheduleOnMain(() -> { // Paper - push to main
+             * this.disconnect(Component.translatable("multiplayer.disconnect.out_of_order_chat"));
+             * }); // Paper - push to main
+             * return false;
+             */
         } else {
             if (this.isChatExpired(instant)) {
                 ServerGamePacketListenerImpl.LOGGER.warn("{} sent expired chat: '{}'. Is the client/server system time unsynchronized? c: {} s: {}", this.player.getName().getString(), s, instant.getEpochSecond(), Instant.now().getEpochSecond()); // Paper
             }
-
-            return this.resetLastActionTime();
         }
+        return this.resetLastActionTime();
+        // Obamium end
     }
 
     private boolean isChatExpired(Instant timestamp) {
